modules/openapi-generator/src/main/resources/python/python-experimental/model_utils.mustache

```diff
@@ -609,6 +612,35 @@ def deserialize_model(model_data, model_class, path_to_item, check_type,
         instance = used_model_class(**kw_args)
     return instance
 
+recursive_discriminators_cache = {}
+
+def recursive_discriminators(model_class):
+    """
+    Return the discriminator mapping for all possible subtypes (recursively found down to the leaves)
+    :param model_class:
+    :return:
+    """
+    if model_class in recursive_discriminators_cache:
+        return recursive_discriminators_cache[model_class]
```

Comment:
How are we sure that this model class is valid?
We are not checking it with from_server and model_data.

Reply:
hmm, not sure I understand. The only purpose of this method is to search through the discriminators and add them to the cache - not to return the model class itself.  That is done in `recursive_discriminator_lookup`. Does that make more sense? This is essentially a recursive replacement for `model_class.discriminator`... I could put it into the model_class definition instead... maybe that would be cleaner? unclear.

Reply:
Ah, I understand